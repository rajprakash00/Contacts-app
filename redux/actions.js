//action types
export const UPDATE_USER = "UPDATE_USER";
export const UPDATE_CONTACT = "UPDATE_CONTACT";

//action creators

export const updateUser = (update) => ({
  type: UPDATE_USER,
  payload: update,
});

export const addContact = (newContact) => ({
  type: UPDATE_CONTACT,
  payload: newContact,
});
