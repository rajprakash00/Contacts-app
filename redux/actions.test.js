import * as actions from "./actions";

describe("updateUser returns actions", () => {
  it("updates the user", () => {
    expect(actions.updateUser({ name: "abcd k" })).toMatchSnapshot();
  });

  it("handles empty name", () => {
    expect(actions.updateUser({ name: "" })).toMatchSnapshot();
  });

  it("handles empty object", () => {
    expect(actions.updateUser({})).toMatchSnapshot();
  });
});

describe("logInUser returns actions", () => {
  const token = "thisIsATestToken";
  const errMessage = "incorrect creds";
  const mockLogin = (username, password) => {
    if (username === "a" && password === "b") {
      return token;
    }
    throw new Error(errMessage);
  };

  it("dipatches LOG_IN_SENT", async () => {
    const mockDispatch = jest.fn();
    await actions.logInUser("", "")(mockDispatch);
    //console.log(mockDispatch.mock.calls[0][0]);
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: actions.LOG_IN_SENT,
    });
  });

  it("dispatches login success with token", async () => {
    const mockDispatch = jest.fn();
    await actions.logInUser("a", "b", mockLogin)(mockDispatch);

    expect(mockDispatch.mock.calls[1]).toMatchSnapshot();
  });

  it("dispatches login failure with incorrect creds", async () => {
    const mockDispatch = jest.fn();
    await actions.logInUser("", "", mockLogin)(mockDispatch);

    expect(mockDispatch.mock.calls[1]).toMatchSnapshot();
  });
});
