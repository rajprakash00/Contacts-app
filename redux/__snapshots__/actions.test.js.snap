// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logInUser returns actions dispatches login failure with incorrect creds 1`] = `
Array [
  Object {
    "payload": "fetch is not defined",
    "type": "LOG_IN_FAILED",
  },
]
`;

exports[`logInUser returns actions dispatches login success with token 1`] = `
Array [
  Object {
    "payload": "fetch is not defined",
    "type": "LOG_IN_FAILED",
  },
]
`;

exports[`updateUser returns actions handles empty name 1`] = `
Object {
  "payload": Object {
    "name": "",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions handles empty object 1`] = `
Object {
  "payload": Object {},
  "type": "UPDATE_USER",
}
`;

exports[`updateUser returns actions updates the user 1`] = `
Object {
  "payload": Object {
    "name": "abcd k",
  },
  "type": "UPDATE_USER",
}
`;
